modules/assim.batch/R/pda.neff.R

```diff
@@ -0,0 +1,163 @@
+      y[i] ~ ddexp(x[i],alpha_0 + alphas[i]*x[i])
+    }
+    
+    #### Process Model
+    for(i in 2:n){
+      x[i] ~ dnorm(x[i-1], tau_add)
+    }
+    
+    #### Priors
+    x[1] ~ dnorm(x_ic,tau_ic)
+    tau_add ~ dgamma(a_add,r_add)
+    }
+    "
+    
+    obs     <- c(input$obs)
+    alpha_0 <- input$par[[1]] 
+    alpha_p <- input$par[[2]] 
+    alpha_n <- input$par[[3]]
+    
+    # converting back C fluxes from  kg C m-2 s-1 to umol C m-2 s-1 
+    # reduces the code and makes model fitting easier
+    if(input$variable.id %in% c(297, 1000000042)){
+      obs <- misc.convert(obs, "kg C m-2 s-1", "umol C m-2 s-1")
+      AMFq  <- rep(0, length(obs))
+      flags <- TRUE
+      AMF.params <- flux.uncertainty(obs, AMFq, flags, bin.num = 20)
+      alpha_0 <- AMF.params$intercept[[1]]
+      alpha_p <- AMF.params$slopeP[[1]]
+      alpha_n <- AMF.params$slopeN[[1]]
+    }
+    
+    data <- list(y = obs, n = length(obs), 
+                 x_ic = 1, tau_ic = 1e-2, a_add = 1, r_add = 1, 
+                 alpha_0 = alpha_0, 
+                 alpha_p = alpha_p, 
+                 alpha_n = alpha_n) 
+    
+    nchain = 3
+    init <- list()
+    for(i in 1:nchain){
+      y.samp    <- sample(obs, length(obs), replace=TRUE)
+      init[[i]] <- list(tau_add=1/var(diff(y.samp), na.rm=TRUE)) 
+    }
+    
+    j.model   <- jags.model(file     = textConnection(HLModel),
+                            data     = data,
+                            inits    = init,
+                            n.chains = 3)
+
+    
+    
+  }else{
+    # Gaussian
+    
+    GaussianModel = "
+    model{
+
+    #### Data Model
+    for(i in 1:n){
+      y[i] ~ dnorm(x[i],tau_obs)
+    }
+
+    #### Process Model
+    for(i in 2:n){
+      x[i] ~ dnorm(x[i-1], tau_add)
+    }
+
+    #### Priors
+    x[1] ~ dnorm(x_ic,tau_ic)
+    tau_obs ~ dgamma(a_obs,r_obs)
+    tau_add ~ dgamma(a_add,r_add)
+    }
+    "
+    
+    obs  <- unlist(input$obs)
+    
+    data <- list(y = obs, n = length(obs),
+                 x_ic =  1, tau_ic = 1e-2,
+                 a_add = 1, r_add = 1, 
+                 a_obs = 1, r_obs = 1)
+    
+    nchain = 3
+    init <- list()
+    for(i in 1:nchain){
+      y.samp    <- sample(obs,length(obs),replace=TRUE)
+      init[[i]] <- list(tau_add=1/var(diff(y.samp), na.rm=TRUE), 
+                        tau_obs=1/var(diff(y.samp), na.rm=TRUE))
+    }
+    
+    j.model   <- jags.model(file     = textConnection(GaussianModel),
+                            data     = data,
+                            inits    = init,
+                            n.chains = 3)
+    
+  }
+  
+  jags.out   <- coda.samples (model          = j.model,
+                              variable.names = c("x"),
+                              n.iter         = 5000,
+                              thin           = 100)
+  
+  
+  out <- as.matrix(jags.out)
+  median.out <- apply(out, 2, median)
+  
+  ar  <- acf(median.out)
+  rho <- as.numeric(ar$acf[2])
```

Comment:
We should chat about this -- I'm not exactly sure what this code is doing, and I'd like to know. I'm sure it's correct -- I'm just genuinely curious about the thought process.